### PR TITLE
Sync DNS with a subdomain

### PR DESCRIPTION
**Pull request by BrianMitchL**
This adds the ability to sync DNS with a subdomain. The implementation in this PR is limited to a single subdomain depth (`sub.example.com` vs `sub.sub.example.com`). The intention is to be able to synchronize a subdomain from the command line.

I also fixed a few bugs to make the code functional (at least in my environment).

### CODE REVIEW

**Comment by shlinx** [comment]
Thanks Brian for making it support subdomain. This is awesome.

File: cloudflare_ddns/cloudflare.py [hunk 1]
```diff
@@ -144,8 +148,8 @@ def create_record(self, dns_type, name, content, **kwargs):
         else:
             data['proxied'] = True
         content = self.request(
-            urllib.parse.urljoin(self.api_url, self.zone['id'], self.zone['id'] + '/dns_records'),
-            'put',
+            self.api_url + self.zone['id'] + '/dns_records',
+            'post',
```
Comment (shlinx) [inline cloudflare_ddns/cloudflare.py]: Does 'put' not work for you? Feeling 'put' is more semantic although both work for me.

Comment (shlinx) [inline cloudflare_ddns/cloudflare.py]: Sorry, please ignore. Looked the wrong line thinking it was in update_record method.

**Comment by BrianMitchL** [comment]
😄 Hopefully everything still works as expected!

**Comment by shlinx** [comment]
@BrianMitchL All seem working, for me at least. I'll write some tests when I've got the chance.

**Pull request merged**